dvc/repo/__init__.py

```diff
@@ -487,7 +489,10 @@ def used_objs(  # noqa: PLR0913
             commit_date=commit_date,
             num=num,
         ):
-            for odb, objs in self.index.used_objs(
+            index, targets = index_from_targets(
+                self, targets, with_deps=with_deps, recursive=recursive
+            )
```

Comment:
Opened for discussion. We short-circuit here to not load `repo.index`, and only load Index from targets.